File: src/models/entities/game.ts
import { DataTypes, Model } from 'sequelize';
import storage from 'storage';
import Round from './round';
import Team from './team';
import { definePk } from '$utils';

export enum GameFieldNames {
    GAME_ID = 'GAME_ID',
    START_DATE = 'START_DATE',
    END_DATE = 'END_DATE',
}

class Game extends Model {
    declare gameId: string;
    declare gameName: string;
    declare startDate: Date;
    declare endDate: Date;
    declare rounds: Round[];
    declare teams: Team[];
};

Game.init({
    gameId: definePk(GameFieldNames.GAME_ID),
    gameName: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    startDate: {
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
        field: GameFieldNames.START_DATE,
    },
    endDate: {
        type: DataTypes.DATE,
        field: GameFieldNames.END_DATE,
    },
}, {
    sequelize: storage,
    modelName: 'Game',
    tableName: 'GAMES',
});

export default Game;

File: brain-ring-dao/src/main/kotlin/dgrishkin/dao/entity/Game.kt
package dgrishkin.dao.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "games")
class Game (
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "games_game_id_seq")
    @SequenceGenerator(name = "games_game_id_seq", allocationSize = 1000)
    @Column(name = "game_id")
    override val id: Long? = null,

    @Column(name = "game_name", nullable = false)
    val gameName: String,

    @Column(name = "start_date", nullable = false)
    val startDate: LocalDateTime? = LocalDateTime.now(),

    @Column(name = "end_date")
    var endDate: LocalDateTime? = null,

    @OneToMany(mappedBy = "game")
    val teams: List<Team>? = arrayListOf(),

    @OneToMany(mappedBy = "game")
    val rounds: List<Round>? = arrayListOf(),
) : BaseEntity